src/core/source_buffers/source_buffers_store.ts

```diff
@@ -114,25 +140,93 @@ export default class SourceBuffersStore {
+   * be equal to either one of those three value:
+   *
+   *   - "set": A SourceBuffer has been created. You will in this case also have
+   *     a second key, `value`, which will contain the related
+   *     QueuedSourceBuffer instance.
+   *     Please note that you will need to wait until `this.onSourceBuffersReady()`
+   *     has emitted before updating a native SourceBuffer.
+   *
+   *   - "disabled": The SourceBuffer has been explicitely disabled for this
+   *     type.
+   *
+   *   - "unset": No action has been taken yet for that SourceBuffer.
    *
    * @param {string} bufferType
    * @returns {QueuedSourceBuffer|null}
    */
-  public get(bufferType : IBufferType) : QueuedSourceBuffer<any>|null {
+  public getStatus(bufferType : IBufferType) : { type : "set";
+                                                 value : QueuedSourceBuffer<any>; } |
+                                               { type : "unset" } |
+                                               { type : "disabled" }
+  {
     const initializedBuffer = this._initializedSourceBuffers[bufferType];
-    return initializedBuffer != null ? initializedBuffer :
-                                       null;
+    return initializedBuffer === undefined ? { type: "unset" } :
+           initializedBuffer === null      ? { type: "disabled" } :
+                                             { type: "set",
+                                               value: initializedBuffer };
+  }
+
+  /**
+   * Native SourceBuffers (audio and video) need to all be created before they
+   * can be used.
+   *
+   * From https://w3c.github.io/media-source/#methods
+   *   For example, a user agent may throw a QuotaExceededError
+   *   exception if the media element has reached the HAVE_METADATA
+   *   readyState. This can occur if the user agent's media engine
+   *   does not support adding more tracks during playback.
+   *
+   * This function will return an Observable emitting when each of these
+   * SourceBuffers is either created or disabled.
+   * @return {Observable}
+   */
+  public onSourceBuffersReady() : Observable<void> {
+    if (this._areNativeSourceBuffersReady()) {
+      return observableOf(undefined);
+    }
+    return new Observable(obs => {
+      this._onNativeSourceBufferAddedOrDisabled.push(() => {
+        if (this._areNativeSourceBuffersReady()) {
+          obs.next(undefined);
+          obs.complete();
+        }
+      });
+    });
+  }
+
+  /**
+   * Explicitely set no SourceBuffer for a given buffer type.
+   * A call to this function is needed at least for unused native buffer types
+   * ("audio" and "video"), to be able to emit through `onSourceBuffersReady`
+   * when both of those are set.
+   * @param {string}
+   */
+  public disableSourceBuffer(bufferType : IBufferType) : void {
+    if (this._initializedSourceBuffers[bufferType] !== undefined) {
+      throw new Error("Cannot disable an active SourceBuffer.");
```

Comment:
Does it mean that we cant disable the `video` track during playback?

Reply:
Yes, if it was already created.
Doing so might trigger a QuotaExceededError.
This is documented in the MSE rec, on the note after the step 3 of appendBuffer: https://www.w3.org/TR/media-source/#methods

Reply:
I commented but removed my comment since, as I was talking about adding a SourceBuffer, not removing one.

I think that you cannot remove a SourceBuffer during playback. To check.